contributors/devel/pull-requests.md

```diff
@@ -1,72 +1,101 @@
+
+**Make sure your PR adheres to our best practices. These include following project conventions, making small PRs, and commenting thoroughly. Please read the more detailed section on [Best Practices for Faster Reviews](#best-practices-for-faster-reviews) at the end of this doc.**
+
+## Run Local Verifications
+
+Run these local verifications before you submit your PR.
 
-# Pull Request Process
+* Enable, run, and pass the Kubernetes [pre-commit hook](development.md#define-a-pre-commit-hook) which checks your repo formatting (and more); note that you may want to add these when you're closer to the end of your project, as they take time to run every time you commit
+* Run and pass `make verify` (can take 30-40 minutes)
+* Run and pass `make test`
+* Run and pass `make test-integration`
 
-An overview of how pull requests are managed for kubernetes.
+## Sign the CLA
 
-This document assumes the reader has already followed the
-[development guide](development.md) to set up their environment,
-and understands
-[basic pull request mechanics](https://help.github.com/articles/using-pull-requests).
+You must sign the CLA before your first contribution. [Read more about the CLA.](https://github.com/kubernetes/community/blob/master/CLA.md)
 
+If you haven't signed the CLA before making a PR, the `k8s-ci-robot` will leave a comment with instructions on how to sign the CLA.
 
-# Life of a Pull Request
+# What to Expect When You Submit a PR
 
-Unless in the last few weeks of a milestone when we need to reduce churn and stabilize, we aim to be always accepting pull requests.
+Merging a PR requires the following steps to be completed before the PR will be merged automatically:
 
-Either the [on call](on-call-rotations.md) manually or the [github "munger"](https://github.com/kubernetes/contrib/tree/master/mungegithub) submit-queue plugin automatically will manage merging PRs.
+- Sign the CLA (prerequisite)
+- Make the PR
+- Update the `release-note-label-needed` label
+- Pass all e2e tests
+- Get a `LGTM` from a reviewer
 
-There are several requirements for the submit-queue to work:
-* Author must have signed CLA ("cla: yes" label added to PR)
-* No changes can be made since last lgtm label was applied
-* k8s-bot must have reported the GCE E2E build and test steps passed (Jenkins unit/integration, Jenkins e2e)
+## Update the Release Note Label (and Write Release Notes if Needed)
 
-Additionally, for infrequent or new contributors, we require the on call to apply the "ok-to-merge" label manually.  This is gated by the [whitelist](https://github.com/kubernetes/contrib/blob/master/mungegithub/whitelist.txt).
+Add release notes to any PR with user-visible changes, such as bug-fixes, feature additions, and output format changes.
 
-## Before sending a pull request
+The `release-note-label-needed` label is added to your PR automatically when you create it. There are a few ways to update it.
```

Comment:
"If you don't add release notes in the PR template, the `release-note-label-needed` label is added...."